### PR TITLE
logging activity and error in log files in addition to the console

### PR DESCRIPTION
**Pull request by fabricehong**
### Background
Related to discussion https://github.com/Torantulino/Auto-GPT/discussions/392

Activity logs will help to analyse agents activity and improve them
Errors logs will help collecting errors in order to debug faster

### Changes

- I moved the main:print_to_console function into a proper file named `logger.py`. I kept the signature, but put it in a Logger class that encapsulate the default python library for logging.
- The logging library accept different logging levels, and provide extensible formatter and handling options.
- The previous behaviour (typing simulation) kept but rewritten in order to fit with the logging library (as a console handler).
- Two new handlers have been added: one for INFO level (activity.log), and one for ERROR level (error.log).
- If the logs folder doesn't exist, it will be created. This folder has been added to the .gitignore
- removed all 'debug' arguments and used logging.DEBUG level instead
- added debug_mode activation through program argument '--debug'

### Test Plan

I tested standard logging (INFO), error logging (ERROR), debug logging (DEBUG) with and without speak mode. Works great.

### Change Safety

- [ ] I have added tests to cover my changes
- [x] I have considered potential risks and mitigations for my changes

I didn't find necessary to add tests for logging.

### CODE REVIEW

File: .gitignore
Comment (nponeccop) [inline .gitignore:1]: Rollback .gitignore as it's unrelated. Submit in a separate PR

**Review by nponeccop — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (fabricehong) [reply .gitignore:1]: This PR now creates logs in that directory and shouldn't be commited. That's why I added an extra-line in the .gitignore.

**Review by nponeccop — APPROVED** [review]

**Comment by fabricehong** [comment]
> The conflicts still remain

alright, solved

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@fabricehong There are conflicts again due to the merging of other PRs

**Comment by nponeccop** [comment]
Yes, I'm soft of a blind robot just to make the PRs look sane. An advanced linter, if you wish. It's @Torantulino that inspects what the code is actually doing and whether it's beneficial for the project to merge.

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@fabricehong There are conflicts again. I really hope to push it through today.

**Comment by fabricehong** [comment]
@nponeccop no worries, It was expected due to the number of PR you had to merge.

Ok it's fixed

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@fabricehong Conflicts again. I'm pushing for it to merge before the Batch 3 starts.

**Comment by fabricehong** [comment]
@nponeccop it should be done in 1 hour. I just realized the debug output with typewriter simulation are not working well when the "Typing..." Spinner is going on. We need to use separate functions for the typing simulation versus simple logging.

**Comment by fabricehong** [comment]
@nponeccop ok it's done.

Here is what I added:
- I differentiated the type_writer logging from the normal logging by changer the destination logger inside the logging encapsulation
- logging functions are now more explicit (typewriter_log, error, debug, warn).

If you merge other PR before, I'll probably have conflicts again because this PR touches a lot of code.

**Review by nponeccop — APPROVED** [review]

**Comment by p-i-** [comment]
I've fixed merge conflicts and merged into `staging` branch, which will be later merged into `master`.
Apologies for shoddy workflow and thanks for the PR @fabricehong

**Comment by fabricehong** [comment]
@p-i- What you cherry-picked on staging won't work because it doesn't include the two latest commits of this afternoon. (see my previous comment).

You need to additionnaly cherry-pick those on staging:
1) 127148d14585e896a674173156578a29788c31b0
2) ce4f729df081ed7cd0d98dae6e88e09b1fb9bdcf

**Comment by fabricehong** [comment]
@p-i- To facilitate your task, I opened a new PR on staging with the two missing commits, with conflicts solved: https://github.com/Torantulino/Auto-GPT/pull/975

**Comment by p-i-** [comment]
@fabricehong Sorry, our workflow is in flux right now as we scale up to clear the backlog of PRs. I deleted staging branch, not realizing 975 was against that branch. Could you possibly resubmit it as a PR against master?

**Comment by fabricehong** [comment]
@p-i- no worries, I totally understand your pain when I see the number or PR :D 

Here it is: https://github.com/Torantulino/Auto-GPT/pull/981

**Comment by fabricehong** [comment]
I see there is conflicts again. Let me 5 min..

**Pull request closed without merging**